users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -166,24 +181,145 @@ def filter_based_masking(x, batch_axis, axis, probability_distribution, max_numb
     feature_indices = tf.reshape(tf.cast(feature_indices, tf.int32), [-1])  # shape: (total_masks,)
 
     # Combine batch offsets and feature indices
-    indices = tf.stack([batch_indices, feature_indices], axis=1) # shape: (total_masks, 2)
+    indices = tf.stack([batch_indices, feature_indices], axis=1)  # shape: (total_masks, 2)
 
     # Create a boolean mask initialized to False. shape: (n_batch, n_features)
     mask = tf.tensor_scatter_nd_update(
         tf.zeros((n_batch, n_features), dtype=tf.bool), indices, tf.ones_like(indices[:, 0], dtype=tf.bool)
     )
     # Reshape the mask to match the input tensor shape
-    mask_shape = [tf.shape(x)[i] if i in (batch_axis, axis) else 1 for i in range(ndim)] # shape: (batch, time, feature)
-    mask = tf.reshape(mask, mask_shape)                                          # shape: (batch, time, feature)
+    mask_shape = [
+        tf.shape(x)[i] if i in (batch_axis, axis) else 1 for i in range(ndim)
+    ]  # shape: (batch, time, feature)
+    mask = tf.reshape(mask, mask_shape)  # shape: (batch, time, feature)
 
     x = tf.where(mask, tf.zeros_like(x), x)
 
     return x
 
 
+def get_peak_to_average_ratio(x):
+    import tensorflow as tf
+
+    x = tf.convert_to_tensor(x)
```

Comment:
Is this needed?

Reply:
It is not, because `get_frequency_response` is already converting to a tensor. Or do you mean the entire function?

Reply:
Just this line.